src/java.base/share/classes/java/lang/invoke/IndirectVarHandle.java

```diff
@@ -90,7 +90,9 @@ public VarHandle withInvokeBehavior() {
     @Override
     @ForceInline
     boolean checkAccessModeThenIsDirect(VarHandle.AccessDescriptor ad) {
-        super.checkAccessModeThenIsDirect(ad);
+        if (exact && accessModeType(ad.type) != ad.symbolicMethodTypeExact) {
```

Comment:
Can we add a comment that the detailed UOE is thrown via `directTarget.getMethodHandleUncached`?

Reply:
Actually `VarHandle::checkAccessModeThenIsDirect` can call `isAccessModeSupported` so that this will check properly for `IndirectVarHandle`.